gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/BalancesApiImpl.java

```diff
@@ -69,26 +71,40 @@ public List<Balance> getAddressBalance(String address) throws IbanNotFoundExcept
                 .map(AccountInfo::getAssetCode)
                 .orElseThrow(() -> new IbanNotFoundException(address));
 
-        var builder = Balance.builder()
-            .address(address)
-            .assetId(0L) // default to 0 now, as we don't have assetId in the system yet
-            .assetName(assetName);
+        var builder = accountCache
+                .getAccountInfo(address)
+                .map(info -> {
+                    Balance.AssetOrLiabilityEnum assetOrLiabilityEnum;
```

Comment:
Maybe this small piece about the liability vs asset thing could be extracted into a function.

Reply:
Done